### DISCUSSION TITLE
How to install cross-compiled packages for linux (WSL2) using vcpkg & Visual Studio 2022

### DISCUSSION THREAD

**Discussion by jonverrier** [microsoft/vcpkg#26034] [Q&A] [answered]
Hi,

I have Visual Studio 2022 with the Linux workload working fine on my machine. I can use the CMake build system to build Linux binaries, debug them from VS, and run them successfully from WSL2 (Ubuntu LTS 20.04). 

I now want build a CMake package with external dependencies (in this case gtest). I have the package installed & working for x64-windows. When I execute ./vcpkg install gtest:x64-linux, I get the error 'No CMAKE_C_COMPILER could be found.' in the log file. 

I am obviously missing something in connecting vcpkg to the C++ compiler installed with the VS Linux workload. 

Any advice? 

Thanks

**Comment by Osyotr** [comment]
> `./vcpkg install gtest:x64-linux`

This command should be executed on a target system (WSL in this case). Then you need to pass correct path to `vcpkg.cmake` on a target system into cmake.

I would also suggest you to switch to manifest mode.

**Reply by jonverrier** [reply]
Thanks - have this working per other comments, although seems to considerably reduce the utility of the Linux workload in VS.

**Reply by Osyotr** [reply]
@jonverrier I've made a cmake script that:
1) Clones vcpkg repo
2) Checkouts revision specified in baseline of default registry or builtin baseline
3) (Optionally) Applies patches to that revision
4) Triggers bootstrap of the executable

[bootstrap_vcpkg.cmake.txt](https://github.com/microsoft/vcpkg/files/9291503/bootstrap_vcpkg.cmake.txt) (remove `.txt` from extension)

Just include it before your first `project()` call and use `"${sourceDir}/_path_where_you_cloned_vcpkg/vcpkg/scripts/buildsystems/vcpkg.cmake"` as a toolchain path.

**Reply by jonverrier** [reply]
Thank you very much!

**Comment by AugP** [comment, marked as answer]
Yes, to be clear, vcpkg is a separate tool from VS, so you won't automatically get the cross-compilation story from VS available with vcpkg. You would need to install vcpkg on the target (in this case WSL2) and run commands there. With that said, this is probably an experience Microsoft could improve. It is somewhere on the backlog, but not likely to be addressed anytime soon unfortunately.

**Reply by jonverrier** [reply]
Thank you. Works, but puts me back to having to switch to linux/WSL, which i was hoping to avoid with the Linux workload. Seems to considerably reduce the utility of the VS linux workload, as any large package will have dependencies which you really want to manage with vcpkg. Might go over to VS & raise it there.
